File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int majorityElement = Integer.MIN_VALUE;

        for(int i = 0; i < nums.length; i++) {
            if(!map.containsKey(nums[i])) {
                map.put(nums[i], 1);
            } else {
                map.replace(nums[i], map.get(nums[i]) + 1);
            }
        }

        for(Map.Entry<Integer, Integer> entry: map.entrySet()) {
            if(entry.getValue() > (nums.length / 2)) {
                majorityElement = entry.getKey();
            }
        }

        return majorityElement;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        majorityElement = None

        for num in nums:
            if not count:
                majorityElement = num
                count += 1
            elif num == majorityElement:
                count += 1
            else:
                count -= 1
        
        return majorityElement
